src/Http/GenericResponseSender.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Tempest\Http;
+
+final readonly class GenericResponseSender implements ResponseSender
+{
+    public function send(Response $response): Response
+    {
+        ob_start();
+
+        $this->sendHeaders($response);
+        $this->sendContent($response);
+
+        ob_end_flush();
+
+        return $response;
+    }
+
+    private function sendHeaders(Response $response): void
+    {
+        // TODO: Handle SAPI/FastCGI
+
+        if (headers_sent()) {
+            return;
+        }
+
+        foreach ($response->getHeaders() as $key => $value) {
+            header("{$key}: {$value}");
+        }
+
+        /**
+         * TODO: Is there a reason to manually set the headers
```

Comment:
How do you mean? This http_response_code only sets the response code and no headers? https://www.php.net/manual/en/function.http-response-code.php

Reply:
I wanted to look into whether there was any difference between using this function and assembling the status code header ourselves. Most major routers Symfony / Slim / etc. are building this manually instead of using this function. That may very well be legacy code, but I wanted to confirm.

Reply:
Gotcha, I'm fine either way